### PR TITLE
Refactor application to use Bookstore API for adding, fetching and removing books.

### PR DESCRIPTION
**Pull request by jonas-45**
This pull request refactors the codebase to use the Bookstore API for adding and removing books. I have implemented the following project requirements:

- [ ] Used the Bookstore API as the backend
- [ ] Used fetch API with createAsyncThunk to make API calls
- [ ] Fetched books from the Bookstore API with useEffect when the component has rendered
- [ ] Refactored the add book and remove book features to persist changes in the server
- [ ] Modified the reducer logic to include HTTP requests to the Bookstore API

I have also tested the functionality of the application after refactoring and everything seems to be working as expected. Styling changes were not required for this project.

Thank you!

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,45 +1,84 @@
...
+//     }),
+//     removeBook: (state, action) => {
+//       const id = action.payload;
+//       return {
+//         ...state,
+//         books: state.books.filter((book) => (book.itemId !== id)),
+//       };
+//     },
+//   },
+// });
```
Comment (tresorsawasawa) [inline src/redux/books/booksSlice.js]: - [ ] Kindly make sure you remove all commented and unused code lines from your code base to make it cleaner, professional and for best practices

File: src/routes/Books.js [hunk 2]
```diff
@@ -1,17 +1,21 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 // import { v4 as uuidv4 } from 'uuid';
```
Comment (tresorsawasawa) [inline src/routes/Books.js]: - [ ] Kindly make sure you remove all commented and unused code lines from your code base to make it cleaner, professional and for best practices

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @jonas-45 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jonas-45** [comment]
Thanks @tresorsawasawa. Working on it right away

**Review by danmainah — APPROVED** [review]
Hi @jonas-45 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**